frontend_tests/node_tests/stream_data.js

```diff
@@ -201,6 +201,23 @@ zrequire('stream_data');
     stream_data.update_subscribers_count(sub);
     assert.equal(sub.subscriber_count, 0);
 
+    // verify that checking subscription with bad email is a noop
+    var bad_email = 'notbrutus@zulip.org';
+    global.blueslip.error = function (msg) {
+        assert.equal(msg, "Unknown email for get_user_id: " + bad_email);
+    };
+    global.blueslip.warn = function (msg) {
+        assert.equal(msg, "Bad email passed to user_is_subscribed: " + bad_email);
+    };
+    assert(!stream_data.user_is_subscribed('Rome', bad_email));
```

Comment:
The test is only testing the negative code-path. It should also make sure that the method returns true if a user is subscribed.